src/objects/Statistic.ts

```diff
@@ -0,0 +1,163 @@
+  isPaused = false;
+
+  /** The value when resetting the game stat. Should be an empty-ish value,
+   * e.g. 0, "", [], etc.
+   */
+  resetValue!: U;
+
+  /**
+   * The function that allows subscribing to updates of the statistic value.
+   * @param updateStat - A callback function that will be called when the statistic value is updated.
+   *                     It takes a single parameter, newValue, which represents the new value of the statistic.
+   */
+  subscriber!: (updateStat: (newValue: T) => void) => void;
+
+  /** The function that accumulates the new value into the existing value */
+  accumulator: (newValue: T, existingValues: U) => U;
+
+  /** The function that formats the value for display */
+  formatDisplay!: (value: U) => string;
+
+  /** The widget that displays the stat */
+  get widget() {
+    return createStatWidget({
+      title: this.statName,
+      gameStatStore: this.gameStatStore,
+      parkStatStore: this.parkStatStore,
+      formatDisplay: this.formatDisplay,
+    });
+  }
+
+  constructor(
+    /** The save/load key */
+    key: string,
+
+    /** The display name on the widget*/
+    title: string,
+
+    /** The value when resetting the game stat. Should be an empty-ish value,
+     * e.g. 0, "", [], etc.
+     */
+    resetValue: U,
+    /**
+     * The function that allows subscribing to updates of the statistic value.
+     * @param updateStat - A callback function that will be called when the statistic value is updated.
+     *                     It takes a single parameter, newValue, which represents the new value of the statistic.
+     */
+    subscriber: (updateStat: (newValue: T) => void) => void,
+
+    /** The function that accumulates the new value into the existing value.
+    * Could concatenate by adding, or spreading into a new array, etc.
+    * 
+    * @example 
+    * function accumulateNewRide(newRide: RideStat, existingRides: RideStat[]) {
+        return [...existingRides, newRide];
+    }
+  */
+    accumulator: (newValue: T, oldValue: U) => U,
+
+    /** The function that formats the value for display */
+    formatDisplay: (value: U) => string
+  ) {
+    this.statKey = key;
+    this.statName = title;
+    this.resetValue = resetValue;
+    this.subscriber = subscriber;
+    this.accumulator = accumulator;
+    this.formatDisplay = formatDisplay;
+
+    // Copilot says to bind the updateStat function to the class
+    // so that it can be passed as a callback
+    this.subscriber(this.updateStat.bind(this));
+
+    this.initialize();
+  }
+
+  initialize() {
+    this.initializeGameStatStore();
+    this.initializeParkStatStore();
+
+    // Subscribe to the pause event
+    areStatisticsPaused.subscribe((isPaused) => (this.isPaused = isPaused));
+
+    // Make sure to load the park stat when the map changes
+    context.subscribe("map.changed", () => {
+      this.loadParkStat();
+    });
+  }
+
+  updateStat(newValue: T) {
+    if (this.isPaused) {
```

Comment:
After some testing, `isPaused` was not being initialized correctly when booting up the game after having stats paused. We could change its definition to `isPaused = areStatisticsPaused.get()`, but at that point, why even have a separate variable? Just delete `isPaused` from the file and change this line to this:

```suggestion
    if (areStatisticsPaused.get()) {
```